app/apps/views.js

```diff
@@ -148,3 +151,28 @@ exports.app_details = function (req, res) {
 
   });
 };
+
+
+exports.app_edit = [
+  ensureLoggedIn('/login'),
+  function(req, res, next) {
+    // This is making three requests cuncurrently to get the app details,
+    // the list of bucket and the list of users.
+    // If all of them succeed it then renders `apps/edit.html`
+    app_request = api.get_app(req.params.id);
+    buckets_request = api.list_buckets();
+    users_request = api.list_users();
+
+    Bluebird
+      .all([app_request, buckets_request, users_request])
```

Comment:
You might be able to [do this with native Promises](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all)

Reply:
I tried yesterday with no success, I want to write some tests and then I could play with this again and see if I can make it work with native Promises.